sonar-plugin-api/src/main/java/org/sonar/api/config/PropertyDefinition.java

```diff
@@ -155,44 +164,59 @@ public static Result validate(PropertyType type, @Nullable String value, List<St
       return Result.SUCCESS;
     }
 
-    switch (type) {
-      case BOOLEAN:
-        return validateBoolean(value);
-      case INTEGER:
-      case LONG:
-        return validateInteger(value);
-      case FLOAT:
-        return validateFloat(value);
-      case SINGLE_SELECT_LIST:
-        if (!options.contains(value)) {
-          return Result.newError("notInOptions");
-        }
-      default:
-        return Result.SUCCESS;
-    }
+    EnumMap<PropertyType, Validation> validations = createValidations(options);
+    return validations.getOrDefault(type, aValue -> Result.SUCCESS).validate(value);
   }
 
-  private static Result validateBoolean(String value) {
-    if (!StringUtils.equalsIgnoreCase(value, "true") && !StringUtils.equalsIgnoreCase(value, "false")) {
-      return Result.newError("notBoolean");
-    }
-    return Result.SUCCESS;
+  private static EnumMap<PropertyType, Validation> createValidations(List<String> options) {
+    EnumMap<PropertyType, Validation> validations = new EnumMap<>(PropertyType.class);
```

Comment:
Nice to have : You could even create the validations in only one line by using :
```
validations = new EnumMap<>(ImmutableMap.of(BOOLEAN, validateBoolean(), INTEGER, validateInteger(), etc.));
```